src/python/pants/core/goals/fmt.py

```diff
@@ -195,44 +222,67 @@ class Fmt(Goal):
     subsystem_cls = FmtSubsystem
 
 
+def _get_request_types(
+    fmt_subsystem: FmtSubsystem,
+    union_membership: UnionMembership,
+) -> tuple[tuple[type[FmtTargetsRequest], ...], tuple[type[FmtBuildFilesRequest], ...]]:
+    fmt_target_request_types = union_membership.get(FmtTargetsRequest)
+    fmt_build_files_request_types = union_membership.get(FmtBuildFilesRequest)
+
+    formatters_to_run = determine_specified_tool_names(
+        "fmt",
+        fmt_subsystem.only,
+        fmt_target_request_types,
+        extra_valid_names=(fbfrt.name for fbfrt in fmt_build_files_request_types),
+    )
```

Comment:
Note, as discussed in Slack, we'll likely be making the `black` and `yapf` BUILD file formatters use the name `black`/`yapf`.